device/core/src/iotedged_authentication_provider.ts

```diff
@@ -0,0 +1,131 @@
+import { AuthenticationProvider, TransportConfig, encodeUriComponentStrict } from 'azure-iot-common';
+import { SharedAccessKeyAuthenticationProvider } from './sak_authentication_provider';
+import { RestApiClient } from 'azure-iot-http-base';
+
+// tslint:disable-next-line:no-var-requires
+const packageJson = require('../package.json');
+
+export const WORKLOAD_API_VERSION = '2018-06-28';
+const DEFAULT_SIGN_ALGORITHM = 'HMACSHA256';
+const DEFAULT_KEY_ID = 'primary';
+
+export interface EdgedAuthConfig {
+  workloadUri: string;
+  deviceId: string;
+  moduleId: string;
+  iothubHostName: string;
+  authScheme: string;
+  gatewayHostName?: string;
+  generationId: string;
+}
+
+interface SignRequest {
+  keyId: string;
+  algo: string;
+  data: string;
+}
+
+interface SignResponse {
+  digest: string;
+}
+
+export class IotedgedAuthenticationProvider extends SharedAccessKeyAuthenticationProvider implements AuthenticationProvider {
```

Comment:
>IotedgedAuthenticationProvider [](start = 13, length = 30)

nit: is that the same name as other SDKs? casing looks a bit weird (although i understand it and am not gonna fight it - just looking for consistency)

Reply:
C# SDK has a "signature provider" and the implementation there calls it `HttpHsmSignatureProvider`. I think this is really just Azure IoT Edge specific. "HttpHsm" sounds too broad. I can rename the class to `IotEdgedAuthenticationProvider` if that's better. The reason I used the spelling as it is right now is because the service that implements the signing is called `iotedged`. I'll rename it to `IotEdgedAuthenticationProvider` though.

Reply:
Checked with John. Apparently C calls it "edge" auth module or something along those lines. I'll align with that and call it `IotEdgeAuthenticationProvider`.